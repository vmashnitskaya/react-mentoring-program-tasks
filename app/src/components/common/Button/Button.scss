@import "../../../constants";

button {
  font-size: 20px;
  border-radius: $main-border-radius;
  outline: none;
  border: none;
  cursor: pointer;
}
