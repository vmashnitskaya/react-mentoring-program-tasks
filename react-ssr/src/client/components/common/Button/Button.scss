@import "src/client/components/constants";

button {
  font-size: 20px;
  border-radius: $main-border-radius;
  outline: none;
  border: none;
  cursor: pointer;
}
