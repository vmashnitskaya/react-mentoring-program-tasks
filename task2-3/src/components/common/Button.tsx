import React, { FunctionComponent, MouseEvent } from 'react';
import './Button.scss';

interface ButtonProps {
    text: string;
    submit?: boolean;
    className?: string;
    value?: string;
    onClick?: (eventPerformed: MouseEvent<HTMLButtonElement>) => void;
}

const Button: FunctionComponent<ButtonProps> = ({
    text,
    submit,
    className,
    value,
    onClick,
}): JSX.Element => {
    const handleClick = (event: MouseEvent<HTMLButtonElement>) => {
        const eventPerformed = event;

        if (onClick) {
            onClick(eventPerformed);
        }
    };

    return (
        <button
            className={`${className} uppercase`}
            type={submit ? 'submit' : 'button'}
            data-value={value}
            onClick={handleClick}
        >
            {text}
        </button>
    );
};

export default Button;
