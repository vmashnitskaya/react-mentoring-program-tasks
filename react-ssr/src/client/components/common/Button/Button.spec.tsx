import React from 'react';
import { render } from '@testing-library/react';
import Button from "./Button";

describe('Button', () =>{
    test('renders button snapshot', () => {
        const {asFragment} = render(<Button text={"Button"}/>);

        expect(asFragment()).toMatchSnapshot();
    });
});
