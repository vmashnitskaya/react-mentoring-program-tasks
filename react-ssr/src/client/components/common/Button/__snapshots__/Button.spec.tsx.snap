// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders button snapshot 1`] = `
<DocumentFragment>
  <button
    class="uppercase"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;
